js/gdm/loginDialog.js

```diff
@@ -778,11 +908,13 @@ const LoginDialog = new Lang.Class({
     cancel: function() {
         if (this._verifyingUser)
             this._userVerifier.cancel();
-        else
-            this._reset();
+        this._reset();
     },
 
     _showPrompt: function(forSecret) {
+        if (this._holdForAnswer)
+            throw new Error("Previous _showPrompt not yet finished");
```

Comment:
Are the callers of `_showPrompt()` prepared to deal with it throwing an error now?